### ISSUE TITLE
NET-1680 Fix S2368 FN: inline arrays

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8100]
An array of an [inline array](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#inline-arrays), while not being explicitly defined as a multidimensional or jagged array, has the same memory layout of those. Moreover, it is accessed in the same way (via double `[]`), and can be as "challenging for developers" (as mentioned by the RSPEC of [S2368](https://sonarsource.github.io/rspec/#/rspec/S2368)) and unintuitive as normal multidimensional arrays (c.p. [this discussion](https://github.com/SonarSource/sonar-dotnet/pull/8082#discussion_r1338542328)).

For these reasons, [S2368](https://sonarsource.github.io/rspec/#/rspec/S2368) should report on them.

```cs
public void AMethod1(Buffer[] a) { }          // FN, Buffer[] is 2-dimensional
public void AMethod2(params Buffer[] a) { }   // Compliant, params of Buffer
public void AMethod3(params Buffer[][] a) { } // Noncompliant, params of Buffer[]
```

Related issues: 
- https://github.com/SonarSource/sonar-dotnet/issues/8083

**Comment by antonioaversa** [comment]
Testing of this issue requires the UnitTest project to work with .NET 8, that is currently not possible due to issue described in this [Trello card](https://trello.com/b/1Zdg1PuD/net-8-c-12-spec-sonarway).

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1680](https://sonarsource.atlassian.net/browse/NET-1680)

[NET-1680]: https://sonarsource.atlassian.net/browse/NET-1680?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
